dev/src/overmind/state.tsx

```diff
@@ -187,7 +207,96 @@ export const state: State = {
     courses: [],
     courseSubmissions: [],
     courseGroupSubmissions: {},
+    sortedAndFilteredSubmissions: derived((state: State, rootState: Context["state"]) => {
+        // Filter and sort submissions based on the current state
+        if (!state.activeCourse || !state.courseSubmissions[state.activeCourse]) {
+            return []
+        }
+        let submissions = state.groupView
+            ? state.courseGroupSubmissions[state.activeCourse]
+            : state.courseSubmissions[state.activeCourse]
+
+        if (!submissions) {
+            return []
+        }
 
+        // If a specific assignment is selected, filter by that assignment
+        const numAssignments = rootState.review.assignmentID > 0
+            ? 1
+            : state.assignments[state.activeCourse].length ?? 0
+
+        let filteredSubmissions = submissions
+        for (const filter of state.submissionFilters) {
+            switch (filter) {
+                case "teachers":
+                    filteredSubmissions = filteredSubmissions.filter(submission => {
+                        return submission.enrollment ? !isTeacher(submission.enrollment) : false
+                    })
+                    break
+                case "approved":
+                    // approved filters all entries where all assignments have been approved
+                    filteredSubmissions = filteredSubmissions.filter(submission => {
+                        if (rootState.review.assignmentID > 0) {
+                            // If a specific assignment is selected, filter by that assignment
+                            const sub = submission.submissions?.find(submission => submission.getAssignment()?.getId() === rootState.review.assignmentID)?.getSubmission()
```

Comment:
Does this return a submission `sub` that is then used as a boolean below? Maybe also use an extra line for the predicate evaluation.

Reply:
The conditional checks the truthy value of `sub` and short circuits if `.find(...)` returns `undefined`, i.e. it does not exist.

As `sub` can only be of the type `undefined` or `Submission`, this is shorthand for `sub !== undefined && isApproved(sub)`.